### PR TITLE
Final cleanups

### PR DESCRIPTION
**Pull request by ikitommi**
* `:form` auto-resolution for core-predicates with `create-spec`
* rename `:pred` => `:spec` in Specs. We wrap specs
* more tests

```clj
(testing "creation"
  (testing "succeeds"
    (is (= (st/spec integer?)
           (st/spec {:spec integer?})
           (st/spec {:spec integer?, :type :long})
           (st/spec integer? {:type :long})
           (st/create-spec {:spec integer?})
           (st/create-spec {:spec integer? :type :long})
           (st/create-spec {:spec integer?, :form `integer?})
           (st/create-spec {:spec integer?, :form `integer?, :type :long}))))

  (testing "anonymous functions"
    (testing "fails without :form"
      (is (thrown?
            #?(:clj Exception, :cljs js/Error)
            (st/create-spec
              {:name "positive?"
               :spec (fn [x] (pos? x))}))))
    (testing ":form and :type can be provided"
      (is (not
            (nil?
              (st/spec?
                (st/create-spec
                  {:name "positive?"
                   :spec (fn [x] (pos? x))
                   :type :long
                   :form `(fn [x] (pos? x))}))))))))
```

### CODE REVIEW

File: src/spec_tools/core.cljc [hunk 1]
```diff
@@ -117,7 +128,9 @@
         "Can't invoke spec with a non-function predicate: " spec)
       {:spec spec})))
 
-(defrecord Spec [pred form type]
+(declare spec)
```
Comment (miikka) [inline src/spec_tools/core.cljc]: Is this declaration needed?

Comment (ikitommi) [inline src/spec_tools/core.cljc]: no, didn't don't that. thanks & fixed.

**Comment by ikitommi** [comment]
semi-dirty, but visible fix to this. Good enough @miikka ?

File: src/spec_tools/visitor.cljc [hunk 2]
```diff
@@ -55,7 +68,7 @@
 
 (defmethod visit 'clojure.spec/every [spec accept]
   (let [[_ inner-spec & kwargs] (formize spec)]
-    (accept 'clojure.spec/every spec [(visit inner-spec accept)])))
+    (accept 'clojure.spec/every spec [(visit (++expand-every-cljs-spec-bug++ inner-spec) accept)])))
```
Comment (miikka) [inline src/spec_tools/visitor.cljc]: Does this work with something like `(s/every (fn [x] (integer? x)))`? If it does, I'm fine with it. 

Did you find the bug on CLJS JIRA? Could you report it if it hasn't been reported yet? I thought it was, but can't find it.

Comment (ikitommi) [inline src/spec_tools/visitor.cljc]: oh, that. well, it does, kinda:

```clj
(jsc/to-json (s/every (fn [x] (integer? x))))
; clj: {:type array, :items {:type integer}}
; cljs: {:type array, :items {}}
```

Comment (ikitommi) [inline src/spec_tools/visitor.cljc]: should not work in either case, but does accidentally(?) with clj.

**Pull request merged**